File: y11599/src/models/AsyncTask.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const AsyncTask = sequelize.define('AsyncTask', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  taskId: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true,
    comment: '任务ID（Bull的Job ID）'
  },
  taskType: {
    type: DataTypes.ENUM('IMPORT_CHANGE_ORDERS', 'IMPORT_REVIEWS', 'IMPORT_REFERENCES', 'IMPORT_SNAPSHOTS', 'EXPORT_DATA', 'DESENSITIZE_DATA', 'BATCH_PROCESS'),
    allowNull: false,
    comment: '任务类型'
  },
  status: {
    type: DataTypes.ENUM('PENDING', 'PROCESSING', 'WAITING_RETRY', 'WAITING_MANUAL', 'FAILED_PERMANENT', 'COMPLETED'),
    defaultValue: 'PENDING',
    comment: '任务状态：待处理/处理中/等待重试/等待人工/永久失败/已完成'
  },
  priority: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '优先级'
  },
  payload: {
    type: DataTypes.TEXT,
    comment: '任务载荷（JSON）'
  },
  batchId: {
    type: DataTypes.STRING,
    comment: '批次ID'
  },
  totalCount: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '总记录数'
  },
  successCount: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '成功记录数'
  },
  failedCount: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '失败记录数'
  },
  skippedCount: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '跳过记录数'
  },
  retryCount: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '已重试次数'
  },
  maxRetries: {
    type: DataTypes.INTEGER,
    defaultValue: 3,
    comment: '最大重试次数'
  },
  errorMessage: {
    type: DataTypes.TEXT,
    comment: '错误信息'
  },
  errorStack: {
    type: DataTypes.TEXT,
    comment: '错误堆栈'
  },
  failedItems: {
    type: DataTypes.JSON,
    defaultValue: [],
    comment: '失败项详情'
  },
  result: {
    type: DataTypes.TEXT,
    comment: '任务结果（JSON）'
  },
  createdBy: {
    type: DataTypes.STRING,
    comment: '创建人ID'
  },
  createdByName: {
    type: DataTypes.STRING,
    comment: '创建人姓名'
  },
  startedAt: {
    type: DataTypes.DATE,
    comment: '开始时间'
  },
  completedAt: {
    type: DataTypes.DATE,
    comment: '完成时间'
  },
  lastRetryAt: {
    type: DataTypes.DATE,
    comment: '最后重试时间'
  },
  nextRetryAt: {
    type: DataTypes.DATE,
    comment: '下次重试时间'
  },
  progress: {
    type: DataTypes.INTEGER,
    defaultValue: 0,
    comment: '进度百分比'
  },
  extra: {
    type: DataTypes.JSON,
    comment: '扩展字段'
  }
}, {
  tableName: 'async_tasks',
  timestamps: true,
  paranoid: true,
  indexes: [
    { fields: ['taskId'] },
    { fields: ['taskType'] },
    { fields: ['status'] },
    { fields: ['batchId'] },
    { fields: ['createdBy'] },
    { fields: ['createdAt'] }
  ]
});

module.exports = AsyncTask;

File: y11533/app/models/async_task.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, DateTime, Text, Boolean, Index

from app.database import Base

class AsyncTask(Base):
    __tablename__ = "async_tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(String(100), unique=True, index=True, comment="任务ID")
    task_type = Column(String(50), nullable=False, comment="任务类型：import/export/replay/reconcile")
    task_name = Column(String(200), comment="任务名称")
    status = Column(String(50), default="pending", comment="任务状态：pending/running/success/waiting_retry/waiting_manual/failed")
    priority = Column(Integer, default=0, comment="优先级")
    
    failure_category = Column(String(50), comment="失败分类：WAITING_RETRY/WAITING_MANUAL/PERMANENT_FAILED")
    failure_reason = Column(Text, comment="失败原因")
    retry_count = Column(Integer, default=0, comment="重试次数")
    max_retry_count = Column(Integer, default=3, comment="最大重试次数")
    next_retry_time = Column(DateTime, comment="下次重试时间")
    
    manual_opinion = Column(Text, comment="人工处理意见")
    manual_operator = Column(String(100), comment="人工处理人")
    manual_time = Column(DateTime, comment="人工处理时间")
    
    input_data = Column(Text, comment="输入数据(JSON)")
    output_data = Column(Text, comment="输出数据(JSON)")
    error_trace = Column(Text, comment="错误堆栈")
    
    progress = Column(Integer, default=0, comment="进度百分比")
    progress_message = Column(String(500), comment="进度消息")
    
    created_by = Column(String(100), comment="创建人")
    created_at = Column(DateTime, default=datetime.now)
    started_at = Column(DateTime, comment="开始时间")
    completed_at = Column(DateTime, comment="完成时间")
    
    parent_task_id = Column(String(100), comment="父任务ID")
    is_resumable = Column(Boolean, default=True, comment="是否可恢复")
    checkpoint = Column(Text, comment="检查点数据(JSON)")
    
    __table_args__ = (
        Index('idx_status_priority', 'status', 'priority'),
        Index('idx_next_retry_time', 'next_retry_time'),
    )
